python/mrtrix3/commands/5ttgen/odf.py

```diff
@@ -0,0 +1,95 @@
+# HSVS SCRIPT
+# Copyright (c) 2008-2024 the MRtrix3 contributors.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# Covered Software is provided under this License on an "as is"
+# basis, without warranty of any kind, either expressed, implied, or
+# statutory, including, without limitation, warranties that the
+# Covered Software is free of defects, merchantable, fit for a
+# particular purpose or non-infringing.
+# See the Mozilla Public License v. 2.0 for more details.
+#
+# For more details, see http://www.mrtrix.org/.
+
+import os
+import subprocess
+from mrtrix3 import MRtrixError
+from mrtrix3 import app, path, run
+
+
+def usage(base_parser, subparsers):  # pylint: disable=unused-variable
+    parser = subparsers.add_parser('odf', parents=[base_parser])
+    parser.set_author("Arkiev D'Souza (arkiev.dsouza@sydney.edu.au) & Robert E. Smith (robert.smith@florey.edu.au)")
+    parser.set_synopsis('Generate a 5TT image from ODF images.')
+    
+    parser.add_argument('template_odf_wm', type=app.Parser.ImageIn(), help='The input white-matter ODF')
+    parser.add_argument('template_odf_gm', type=app.Parser.ImageIn(), help='The input grey-matter ODF')
+    parser.add_argument('template_odf_csf', type=app.Parser.ImageIn(), help='The input cerebrospinal fluid ODF')
+    parser.add_argument('mask_image', type=app.Parser.ImageIn(), help='The input binary brain mask image')
+    parser.add_argument('fTT_image', type=app.Parser.ImageOut(), help='The output 5TT image')
+
+    options = parser.add_argument_group('Options specific to the "odf" algorithm')
+    options.add_argument('-sgm', nargs='?', default=None, type=app.Parser.ImageIn(), help='Provide a mask of voxels that should be set to sub-cortical grey matter')
+    options.add_argument('-pathology', nargs='?', default=None, type=app.Parser.ImageIn(), help='Provide a mask of voxels that should be set to pathological tissue')
+
+
+def check_file(filepath):
+    if not os.path.isfile(filepath):
+        raise MRtrixError(f'Required input file missing (expected location: {filepath})')
+
+
+def execute():  # pylint: disable=unused-variable
+    # Check for input files
+    check_file(app.ARGS.template_odf_wm)
+    check_file(app.ARGS.template_odf_gm)
+    check_file(app.ARGS.template_odf_csf)
+    check_file(app.ARGS.mask_image)
```

Comment:
```suggestion
def execute(): # pylint: disable=unused-variable
```

On `dev`, this type of input / output file checking is no longer done explicitly by the developer, it's instead handled by the typed CLI. 

In this case it's actually worse; doing this check would prevent valid use cases. One of these inputs could be a using the multi-file numbered image format. In this case, there will not exist a filesystem entry corresponding to the string specified at the command-line, yet a valid input image can nevertheless be read once that string is parsed by the C++ API.